Sources/Workspace/InitPackage.swift

```diff
@@ -265,6 +265,8 @@ public final class InitPackage {
     private func writeLinuxMain(testsPath: AbsolutePath) throws {
         try writePackageFile(testsPath.appending(component: "LinuxMain.swift")) { stream in
             stream <<< """
+                #if !os(watchOS)
```

Comment:
Another option is `canImport`, which would be a little more semantic, but this would be fine too.

Reply:
This is such a random check :/ Will `#if canImport(XCTest)` work?

Reply:
Actually, I don't think this is the right approach. If there is no XCTest on watchOS then the build system should ignore test targets.

Reply:
I don't think that's reasonable. Test targets should not be inherently tied to XCTest.

Reply:
> Test targets should not be inherently tied to XCTest.

Maybe that could be a long term goal. But right now if you write a `.testTarget` without importing XCTest, then `swift test` and `xcodebuild test` simply run 0 tests. Do you want watchOS to say “Testing passed.” when nothing was actually tested?

> If there is no XCTest on watchOS then the build system should ignore test targets.

That seems like the right solution to me.

In the same way it would also be nice to ignore executable targets on iOS, etc. Right now (5.0.1) adding an executable target breaks the entire package for those platforms (in the generated Xcode project).

Reply:
I also just tried adding a `main.swift` to a test target to see if it is possible to evade XCTest that way. It doesn’t build:

```shell
Tests/PackageTests/main.swift:2:1: error: expressions are not allowed at the top level
print("Testing...")
^
```

Reply:
Agreed. We should do something better in the long run. But the build system should help us as long as we're solely using XCTest for our test targets.

Reply:
The result will be the same. If the build system ignores test targets, we also end up running 0 tests and the result will be passing.